test/test_init_controller.py

```diff
@@ -0,0 +1,62 @@
+import io
+import unittest
+from unittest.mock import patch, Mock, MagicMock
+
+import OpenSSL
+from OpenSSL import SSL
+
+from conjur import init_controller, init_logic
+from conjur.init_logic import InitLogic
+from conjur.input_conjurrc import InputConjurrc
+
+
+class InitControllerTest(unittest.TestCase):
+    capture_stream = io.StringIO()
+    mock_input_object = InputConjurrc()
+    init_logic = InitLogic()
+
+    '''
+    Tests no account provided by user raises Runtime error
+    '''
+    @patch('builtins.input', return_value=None)
+    def test_init_without_host_raises_error(self, mock_input):
+        with self.assertRaises(RuntimeError):
+            self.mock_input_object.appliance_url = 'https://someurl'
+            init_controller.InitController.get_host_info(self.mock_input_object)
+
+    # test_init_without_host_raises_error.tester=True
+
+    '''
+    Tests that an exception will be raised when user does not trust certificate
+    '''
+    @patch('builtins.input', side_effect=['no'])
+    def test_init_not_trusting_cert_raises_error(self, mock_input):
+        self.mock_input_object.appliance_url = 'https://someurl'
+
+        ctx = SSL.Context(method=SSL.TLSv1_2_METHOD)
+        sock = OpenSSL.SSL.Connection(ctx)
+        self.init_logic.connect_to_socket = MagicMock(return_value = sock)
+        self.init_logic.get_certificate = MagicMock(return_value = ["12:AB", "somecertchain"])
+
+        with self.assertRaises(ValueError):
+            init_controller.InitController.get_server_certificate(self.init_logic, self.mock_input_object)
+
+    # test_init_not_trusting_cert_raises_error.tester=True
+
+    '''
+    Tests that an certificate was returned when user trusts the certificate
+    '''
+    @patch('builtins.input', side_effect=['yes'])
+    def test_init_user_trusts_cert_returns_cert(self, mock_input):
```

Comment:
This test file should include the same test but with multiple certs too

Reply:
Isn't it enough to add `mock_certificate = "-----BEGIN CERTIFICATE-----\n-----BEGIN CERTIFICATE-----"`?

Reply:
What would multiple certs in a test validate for us?

Reply:
That the code can correctly write them out to file when the server returns them. This is an edge case but it's decently common.

Reply:
This can be an integration test. Will add